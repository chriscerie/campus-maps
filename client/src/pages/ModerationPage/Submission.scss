.acceptance-buttons-container {
  text-align: right;
}

.acceptance-buttons {
  padding-left: 50px;
}

.submission-link {
  color: #2592ff;
  text-decoration: none;
}

.submission-link:hover {
  color: #2537ff;
}

.submissions-container {
  margin-bottom: 1.5em;
}

.submission-divider {
  width: 90%;
}

.submission-header-container {
  display: table;
  border-spacing: 1.5em 0;
  vertical-align: middle;
  margin: 0 -1.5em 1.5em;
  margin-top: 1.5em;
}

.submission-header-body {
  display: table;
  border-collapse: separate;
  border-spacing: 1.5em 0;
  margin: 0 -1.5em;
}

.submission-profile-image-container {
  display: table-cell;
  border-radius: 50%;
  overflow: hidden;
}

.submission-profile-image {
  height: 4em;
  width: 4em;
  aspect-ratio: auto 1;
  vertical-align: middle;
  box-sizing: border-box;
}

.submission-location-info-container {
  display: table-cell;
  vertical-align: middle;
  width: 100%;
}

.submission-location-info {
  display: block;
  font-size: larger;
  font-weight: 700;
}

.submission-text-container {
  width: 90%;
}

.submission-text {
  margin: 0;
  font-size: smaller;
  line-height: 1.7em;
}

#submission-accept-button {
  color: #fff;
  background-color: #2592ff;
  margin-left: 5px;
  margin-right: 5px;
  border: 1px solid #2592ff;
  padding: 0.4em 1.4em;
  border-radius: 0.3em;
  font-weight: bold;
  font-size: 1em;
  text-transform: none;
}

#submission-accept-button:hover {
  background-color: #2537ff;
  border: 1px solid #2537ff;
}

#submission-reject-button {
  color: #929292;
  background-color: #fff;
  margin-left: 5px;
  margin-right: 5px;
  border: 1px solid #929292;
  padding: 0.4em 1.4em;
  border-radius: 0.3em;
  font-weight: bold;
  font-size: 1em;
  text-transform: none;
}

#submission-reject-button:hover {
  background-color: #929292;
  color: #fff;
}
