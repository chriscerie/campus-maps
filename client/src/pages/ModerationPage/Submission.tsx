import { Button, Divider } from '@mui/material';
import type { UserType } from '../../types/UserType';
import { Fragment } from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import './Submission.scss';

export type SubmissionType = {
  _id: string;
  author_id: string;
  id: string;
  name: string;
  type: string;
  description: string;
  address1: string;
  address2: string;
  city: string;
  state: string;
  zip_code: string;
  rooms: Array<{
    room_name: string;
    room_id: string;
  }>;
  __v: string;
  author?: UserType;
};

const Submission = ({
  submission,
  resetSubmissions,
}: {
  submission: SubmissionType;
  resetSubmissions: () => void;
}) => {
  return (
    <Fragment>
      <li className="submissions-container">
        <Divider className="submission-divider" />

        <div className="submission-header-container">
          <div>
            <div className="submission-header-body">
              <div className="submission-profile-image-container">
                <img
                  src={
                    submission.author ? submission.author.profile_picture : ''
                  }
                  alt="profile"
                  className="submission-profile-image"
                />
              </div>
              <div className="submission-location-info-container">
                <div className="submission-location-info">
                  <Link
                    className="submission-link"
                    to={`/loc/${submission.id}`}
                  >
                    {submission.name}
                  </Link>
                </div>
                <div>
                  Update by {submission.author?.first_name}{' '}
                  {submission.author?.last_name}
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className="submission-text-container">
          <p className="submission-text">
            <strong>Type:</strong> {submission.type} <br />
            <strong>Description:</strong> {submission.description} <br />
            <strong>Address 1:</strong> {submission.address1} <br />
            <strong>Address 2:</strong> {submission.address2} <br />
            <strong>City:</strong> {submission.city} <br />
            <strong>State:</strong> {submission.state} <br />
            <strong>ZIP Code:</strong> {submission.zip_code} <br />
            <strong>Rooms:</strong> <br />
            {submission.rooms &&
              submission.rooms.map((room) => (
                <div>
                  {'â€¢ ' + room.room_name}
                  <br />
                </div>
              ))}
          </p>
          <div className="acceptance-buttons-container">
            {/* Buttons are nonfunctional currently */}
            <Fragment>
              <Button
                variant="contained"
                disableElevation
                id="submission-accept-button"
                onClick={() => {
                  axios
                    .post(`/api/v1/locations/moderation/${submission._id}`, {
                      decision: 'Accept',
                    })
                    .then(() => {
                      resetSubmissions();
                    });
                }}
              >
                Accept
              </Button>
              <Button
                variant="contained"
                disableElevation
                id="submission-reject-button"
                onClick={() => {
                  axios
                    .post(`/api/v1/locations/moderation/${submission._id}`, {
                      decision: 'Deny',
                    })
                    .then(() => {
                      resetSubmissions();
                    });
                }}
              >
                Reject
              </Button>
            </Fragment>
          </div>
        </div>
      </li>
    </Fragment>
  );
};

export default Submission;
