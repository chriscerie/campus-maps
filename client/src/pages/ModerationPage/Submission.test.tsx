import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import store from '../../reducers';
import Submission from './Submission';

describe('Submission', () => {
  it('should render successfully', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Submission
              submission={{
                _id: '_id',
                author_id: 'author_id',
                id: 'id',
                name: 'name',
                type: 'type',
                description: 'description',
                address1: 'address1',
                address2: 'address2',
                city: 'city',
                state: 'state',
                zip_code: 'zip_code',
                rooms: [],
                __v: '__v',
              }}
              resetSubmissions={() => {}}
            />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
