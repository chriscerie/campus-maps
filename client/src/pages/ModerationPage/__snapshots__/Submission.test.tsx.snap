// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submission should render successfully 1`] = `
<li
  className="submissions-container"
>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-9mgopn-MuiDivider-root{margin:0;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;border-width:0;border-style:solid;border-color:rgba(0, 0, 0, 0.12);border-bottom-width:thin;}",
      }
    }
    data-emotion="css 9mgopn-MuiDivider-root"
  />
  <hr
    className="MuiDivider-root MuiDivider-fullWidth submission-divider css-9mgopn-MuiDivider-root"
  />
  <div
    className="submission-header-container"
  >
    <div>
      <div
        className="submission-header-body"
      >
        <div
          className="submission-profile-image-container"
        >
          <img
            alt="profile"
            className="submission-profile-image"
            src=""
          />
        </div>
        <div
          className="submission-location-info-container"
        >
          <div
            className="submission-location-info"
          >
            <a
              className="submission-link"
              href="/loc/id"
              onClick={[Function]}
            >
              name
            </a>
          </div>
          <div>
            Update by 
             
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="submission-text-container"
  >
    <p
      className="submission-text"
    >
      <strong>
        Type:
      </strong>
       
      type
       
      <br />
      <strong>
        Description:
      </strong>
       
      description
       
      <br />
      <strong>
        Address 1:
      </strong>
       
      address1
       
      <br />
      <strong>
        Address 2:
      </strong>
       
      address2
       
      <br />
      <strong>
        City:
      </strong>
       
      city
       
      <br />
      <strong>
        State:
      </strong>
       
      state
       
      <br />
      <strong>
        ZIP Code:
      </strong>
       
      zip_code
       
      <br />
      <strong>
        Rooms:
      </strong>
       
      <br />
    </p>
    <div
      className="acceptance-buttons-container"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1diear7-MuiButton-root{font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:6px 16px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#fff;background-color:#1976d2;box-shadow:0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);box-shadow:none;}.css-1diear7-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:#1565c0;box-shadow:0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);}@media (hover: none){.css-1diear7-MuiButton-root:hover{background-color:#1976d2;}}.css-1diear7-MuiButton-root:active{box-shadow:0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);}.css-1diear7-MuiButton-root.Mui-focusVisible{box-shadow:0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);}.css-1diear7-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);box-shadow:none;background-color:rgba(0, 0, 0, 0.12);}.css-1diear7-MuiButton-root:hover{box-shadow:none;}.css-1diear7-MuiButton-root.Mui-focusVisible{box-shadow:none;}.css-1diear7-MuiButton-root:active{box-shadow:none;}.css-1diear7-MuiButton-root.Mui-disabled{box-shadow:none;}",
          }
        }
        data-emotion="css 1diear7-MuiButton-root"
      />
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-zddlty-MuiButtonBase-root-MuiButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:6px 16px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#fff;background-color:#1976d2;box-shadow:0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);box-shadow:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root::-moz-focus-inner{border-style:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-zddlty-MuiButtonBase-root-MuiButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-zddlty-MuiButtonBase-root-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:#1565c0;box-shadow:0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);}@media (hover: none){.css-zddlty-MuiButtonBase-root-MuiButton-root:hover{background-color:#1976d2;}}.css-zddlty-MuiButtonBase-root-MuiButton-root:active{box-shadow:0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-focusVisible{box-shadow:0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);box-shadow:none;background-color:rgba(0, 0, 0, 0.12);}.css-zddlty-MuiButtonBase-root-MuiButton-root:hover{box-shadow:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-focusVisible{box-shadow:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root:active{box-shadow:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-disabled{box-shadow:none;}",
          }
        }
        data-emotion="css zddlty-MuiButtonBase-root-MuiButton-root"
      />
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-zddlty-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        id="submission-accept-button"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Accept
      </button>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1diear7-MuiButton-root{font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:6px 16px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#fff;background-color:#1976d2;box-shadow:0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);box-shadow:none;}.css-1diear7-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:#1565c0;box-shadow:0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);}@media (hover: none){.css-1diear7-MuiButton-root:hover{background-color:#1976d2;}}.css-1diear7-MuiButton-root:active{box-shadow:0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);}.css-1diear7-MuiButton-root.Mui-focusVisible{box-shadow:0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);}.css-1diear7-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);box-shadow:none;background-color:rgba(0, 0, 0, 0.12);}.css-1diear7-MuiButton-root:hover{box-shadow:none;}.css-1diear7-MuiButton-root.Mui-focusVisible{box-shadow:none;}.css-1diear7-MuiButton-root:active{box-shadow:none;}.css-1diear7-MuiButton-root.Mui-disabled{box-shadow:none;}",
          }
        }
        data-emotion="css 1diear7-MuiButton-root"
      />
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-zddlty-MuiButtonBase-root-MuiButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:6px 16px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#fff;background-color:#1976d2;box-shadow:0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);box-shadow:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root::-moz-focus-inner{border-style:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-zddlty-MuiButtonBase-root-MuiButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-zddlty-MuiButtonBase-root-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:#1565c0;box-shadow:0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);}@media (hover: none){.css-zddlty-MuiButtonBase-root-MuiButton-root:hover{background-color:#1976d2;}}.css-zddlty-MuiButtonBase-root-MuiButton-root:active{box-shadow:0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-focusVisible{box-shadow:0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);box-shadow:none;background-color:rgba(0, 0, 0, 0.12);}.css-zddlty-MuiButtonBase-root-MuiButton-root:hover{box-shadow:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-focusVisible{box-shadow:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root:active{box-shadow:none;}.css-zddlty-MuiButtonBase-root-MuiButton-root.Mui-disabled{box-shadow:none;}",
          }
        }
        data-emotion="css zddlty-MuiButtonBase-root-MuiButton-root"
      />
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-zddlty-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        id="submission-reject-button"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Reject
      </button>
    </div>
  </div>
</li>
`;
